### PR TITLE
[Refactor] remove HiveMetaStoreTable concept and add `getDbName` in `Table`

### PR DESCRIPTION
**Pull request by dirtysalt**
## Why I'm doing:

There are so many overlapped concepts and interfaces on tables
- `HiveMetaStoreTable` which basically have all interfaces in `Table`
- we have `getTableName` and `getName` to get table name.
- `Table` should have `getDbName` as a basic method but actually does not have.  And it leads many subclasses to have `getDbName` which is really bad.

## What I'm doing:

This PR is to:
- add `getDbName()` in Table, which alsmost all subclasses need.
- remove `HiveMetaStoreTable` concept from FE.
- do more simplication based on the above simplications

------------

And in the future, we can keep simplifying code like following code

```
  GlobalStateMgr.getCurrentState().getAnalyzeMgr().removeExternalBasicStatsMeta(table.getCatalogName(),
          table.getDbName(), table.getName());
  GlobalStateMgr.getCurrentState().getAnalyzeMgr().dropAnalyzeJob(table.getCatalogName(),
          table.getDbName(), table.getName());
```

Can be rewritten as `table` as parameters which is more elegant and clear

```
  GlobalStateMgr.getCurrentState().getAnalyzeMgr().removeExternalBasicStatsMeta(table);
  GlobalStateMgr.getCurrentState().getAnalyzeMgr().dropAnalyzeJob(table);
```

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java [hunk 1]
```diff
@@ -272,6 +276,10 @@ public String getTableIdentifier() {
         return name;
     }
 
+    public String getDbName() {
```
Comment (murphyatwork) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Table.java:279]: I would suggest make it `Optional<String> mayGetDbName()`, bc some types of table may never support it

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java [hunk 2]
```diff
@@ -176,12 +176,13 @@ public String getCatalogName() {
...
     }
 
+    @Override
     public String getDbName() {
         return hiveDbName;
     }
 
     @Override
-    public String getTableName() {
+    public String getName() {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:185]: missing iceberg table?

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java (hunk 1 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Table.java:279]: So just throw exception if not suported.

Remember there a lot of places that call `getDbName()` directly.

File: fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java (hunk 2 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:185]: @stephen-shelby  you mean

```    public String getResourceName() {
        return resourceName;
    }

    public String getRemoteDbName() {
        return remoteDbName;
    }

    public String getRemoteTableName() {
        return remoteTableName;
    }

```
Changed to getDbName and getName?

Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HiveTable.java:185]: @stephen-shelby can you take a look at my fix?

**Review by stephen-shelby — DISMISSED** [review]

**Pull request closed without merging**